src/main.cpp

```diff
@@ -5084,17 +5050,26 @@ bool ProcessNewBlock(CValidationState& state, CNode* pfrom, CBlock* pblock, CDis
         }
     }
 
+    // Preliminary checks
+    int64_t nStartTime = GetTimeMillis();
+
+    // check block
+    bool checked = CheckBlock(*pblock, state);
+
+    if (!CheckBlockSignature(*pblock))
+        return error("ProcessNewBlock() : bad proof-of-stake block signature");
```

Comment:
__func__

Reply:
Hey, based on the importance of this PR, let's try to focus the review toward the real purpose and not clean main.cpp big scary file.

The file should be cleaned and divided in several files in several PRs, similar organization to what bitcoin has now (would be good to have an open issue for this work).

Reply:
fair enough; just catching things in code that was touched given similar things were changed in this PR as well.  (And it was initial feedback given time constraints before I dug in deeper).  But of course it's one of the reasons I make them as comments only.  

However... msg received :)